lib/bing_ads_ruby_sdk/oauth2/authorization_handler.rb

```diff
@@ -5,25 +5,27 @@ module BingAdsRubySdk
   module OAuth2
     # Adds some useful methods to Signet::OAuth2::Client
     class AuthorizationHandler
-
+      SCOPE = "https://ads.microsoft.com/msads.manage"
+      AUTHORIZE_URI = "https://login.microsoftonline.com/common/oauth2/v2.0/authorize"
+      TOKEN_URI = "https://login.microsoftonline.com/common/oauth2/v2.0/token"
+      REDIRECT_URI = "https://login.microsoftonline.com/common/oauth2/nativeclient"
       # @param developer_token
       # @param client_id
       # @param store [Store]
       def initialize(developer_token:, client_id:, store:, client_secret: nil)
         @client = Signet::OAuth2::Client.new(
           client_params(developer_token, client_id, client_secret)
         )
-        @store  = store
-        refresh_from_store
```

Comment:
when I think about it, its best to delay the read of the token as much as possible as some other thread could have updated it